static/app/views/profiling/landing/chart.tsx

```diff
@@ -0,0 +1,184 @@
+import {useMemo} from 'react';
+import {browserHistory, withRouter, WithRouterProps} from 'react-router';
+import {useTheme} from '@emotion/react';
+import {Location} from 'history';
+
+import ChartZoom from 'sentry/components/charts/chartZoom';
+import OptionSelector from 'sentry/components/charts/optionSelector';
+import ScatterChart from 'sentry/components/charts/scatterChart';
+import {
+  ChartContainer,
+  ChartControls,
+  InlineContainer,
+} from 'sentry/components/charts/styles';
+import TransitionChart from 'sentry/components/charts/transitionChart';
+import TransparentLoadingMask from 'sentry/components/charts/transparentLoadingMask';
+import {getSeriesSelection} from 'sentry/components/charts/utils';
+import {Panel} from 'sentry/components/panels';
+import Placeholder from 'sentry/components/placeholder';
+import {t} from 'sentry/locale';
+import {SelectValue} from 'sentry/types';
+import {EChartClickHandler, SeriesDataUnit} from 'sentry/types/echarts';
+import {Trace} from 'sentry/types/profiling/trace';
+import {defined} from 'sentry/utils';
+import {axisLabelFormatter, tooltipFormatter} from 'sentry/utils/discover/charts';
+import getDynamicText from 'sentry/utils/getDynamicText';
+import {decodeScalar} from 'sentry/utils/queryString';
+
+interface Props extends WithRouterProps {
+  location: Location;
+  traces: Trace[];
+  end?: string;
+  start?: string;
+  statsPeriod?: string | null;
+  utc?: string;
+}
+
+function _ProfileScatterChart({
+  router,
+  location,
+  traces,
+  start,
+  end,
+  statsPeriod,
+  utc,
+}: Props) {
+  const theme = useTheme();
+
+  const colorEncoding = useMemo(() => getColorEncodingFromLocation(location), [location]);
+
+  const series = useMemo(() => {
+    const seriesMap: Record<string, SeriesDataUnit[]> = {};
+
+    for (const row of traces) {
+      const seriesName = row[colorEncoding];
+      if (!seriesMap.hasOwnProperty(seriesName)) {
+        seriesMap[seriesName] = [];
+      }
+      seriesMap[seriesName].push({
+        name: row.start_time_unix * 1000,
+        value: row.trace_duration_ms,
+      });
+    }
+
+    return Object.entries(seriesMap).map(([seriesName, data]) => ({seriesName, data}));
+  }, [colorEncoding]);
+
+  // TODO
+  const onClickHandler: EChartClickHandler = _params => {};
+
+  const loading = false;
+  const reloading = false;
+
+  const chartOptions = {
```

Comment:
I usually try to avoid configurations like these in component body. They make the component less readable and are usually low value to whoever is reading/refactoring this code. Wdyt about having a makeProfilingChartOptions fn that returns these props?